### DISCUSSION TITLE
Battery Mode features disappear when Vantage disabled

### DISCUSSION THREAD

**Discussion by TheDannemand** [BartoszCichecki/LenovoLegionToolkit#1233] [Q&A] [answered]
I came across LLT when looking for a way to avoid Vantage on my new Lenovo Yoga and still be able to set **Battery Mode**.  

First, this software is a thing of beauty: I was blown away by its clean layout, completeness and dynamic feature/module structure. Very impressive. I donated, and will donate again next month.

Second, I completely understand that Yoga is not a supported model and that adding other models than Legion is NOT on the agenda. It sort-of-works on my Yoga, though of course with a lot of missing features (no GPU etc). I mostly use it to change **Battery Mode** (Conservation/Normal/Rapid) and **Overnight Battery Charging**. 

But I am finding that the **Battery Mode** control disappears immediately after being changed UNLESS Vantage is still enabled. As long as Vantage is enabled, LLT works perfectly. But if Vantage is disabled, the **Battery Mode** control only shows when it is set to Conservation. If I change **Battery Mode** (to Normal or Rapid) the control disappears, and I have to re-enable Vantage and use it to change Battery Mode back to Conservation. At that point LLT will show the control again, until next time I change it.

So in practical terms, I only use LLT to disable/enable Vantage -- which is still very useful, although it does mean a lot of rebooting.

Is this a known bug in the current LLT version, or is it caused by my Yoga model being incompatible? If the latter, of course I apologize for wasting your time. Otherwise I can provide a log file during an LLT session where this occurs.

I appreciate any response or feedback. Thank you in advance.

-TheDannemand

PS. I live in USA but am originally from Denmark (hence my username). I see you live in Denmark too, but I think I gather you have Ukrainian roots? I am a fierce Ukraine supporter and am deeply embarrassed by what has been happening with US support for Ukraine. I apologize on behalf of all those Americans who care about freedom rather than politics.

**Comment by agzionvard** [comment]
I actually have the same question. I have a Legion Pro 7i i9 13900HX RTX 4090 and I've only tried LLT a few days ago. I was surprised to know just now that I can't find the battery mode anywhere. The option is not visible even if it's added via the customize button. I'm concerned because my battery is now constantly at 100% for the past few days. I use conversation mode on Vantage and now it's not applying when it's deactivated via LLT. I wonder how this is supposed to work. I'm always plugged in because I don't take this laptop anywhere and I used to have a max of 75%-80% battery via conservation mode, and now it's not working. I hope someone can help.

EDIT: I recently tried a bit of registry editing based on a recommendation on reddit regarding power throttling problems for my model. Unfortunately, it had the side effect of disabling the system devices needed for battery mode. So I just undid what I did in the registry and now the battery mode is visible again. It was the AVCPIF and ipf_cpu. Had to set the Start code back to 3 for enabled.

**Reply by TheDannemand** [reply]
Thank you for you for sharing this. I am somewhat relieved to hear that you have this problem too on a Legion model. It brings me hope that it is fixable in LLT and will (hopefully) work in my Yoga model as well.

For me, LLTs behavior is completely consistent and reproducible: When **Battery Mode=Conservation**, the **Battery Mode** control shows in LLT. When Battery Mode is any other setting (Normal or Rapid), the **Battery Mode** control disappears from LLT. The way to get the control back is to re-enable Vantage and use it to set **Battery Mode=Conservation**. The control will then show next time I load LLT.

Of course one can leave Vantage enabled all the time, but that kind of defeats the purpose of LLT (at least for me) which is to get rid of the bloat, telemetry and spamming of Vantage. If this can be fixed in LLT, I may uninstall Vantage altogether, as LLT has everything I need: Battery Mode, Battery status, even Warranty status. It is an impressive piece of software for sure!

**Reply by BartoszCichecki** [reply, marked as answer]
Vantage isn't needed. LLT asks the energy driver for a status, if the status can't be retrieved the control won't be shown. There can be many reason why this happens, most commonly lack of energy driver or like in the case above, strange changes to the OS. I can't speak for other laptops than Legion though.

**Reply by TheDannemand** [reply]
@BartoszCichecki: Thank you for the reply. And thanks again for this awesome piece of software (to you AND the other contributors). Again, I completely understand about not being able to speak to other models. Seeing as @agzionvard also uses this on a Legion model, can you confirm that you are NOT able to reproduce the issue experienced by us both where the **Battery Mode** control disappears after changing its setting to something OTHER than Conservation?

@agzionvard: Now that you have the **Battery Mode** control back, does it stay if you change its setting?

In my case it is a relatively new factory install. Lenovo power driver is definitely installed: Fn+Q works, even with Vantage disabled.

Thanks again. To you both.

**Reply by agzionvard** [reply]
It stays now. My only error was the registry thing I did which I mentioned prior. Since yours is practically new, I'm sure the registry hasn't been tampered with as I have, so AVCPIF and ipf_cpu are for sure functioning as they are. So I'm not sure what's causing the disappearing LLT battery mode on your end. Have you tried checking in your Device Manager>System Devices and looking for anything amiss there? When I tampered with my registry, there were two devices under System Devices that had a prompt (device not working properly) which corresponded to AVCPIF and ipf_cpu being modified in my registry. Only when I changed it back did the prompts disappear and LLT and Vantage went back to working as they should.

**Reply by TheDannemand** [reply]
@agzionvard: Excellent, thank you for that clarification. Glad it is working consistently for you now. Unfortunately in my case, it is more likely to be the different model, as @BartoszCichecki mentioned and as I originally feared. Again, it is a relatively new factory install, with everything ship-shape (after trimming MS/Win bloat, of course). All clean in Device manager. Event Viewer too, except the usual suspects (including a .NET error tripped by Vantage, relating to display color adjustment, not power).

I'll just use LLT to disable/enable Vantage, for which it is still useful. Fortunately, it is not often I need to change **Battery Mode**, as 80% is plenty for my use most days.

Thank again, both!

**Reply by BartoszCichecki** [reply]
> can you confirm that you are NOT able to reproduce the issue experienced by us both where the Battery Mode control disappears after changing its setting to something OTHER than Conservation?

As I said, if the battery  mode disappears it's due LLT not being able to talk to the energy driver. It can be due to the fact that it isn't installed, or Windows was tweaked in a way that breaks it or, like in your case most likely, incompatibility.

**Reply by BartoszCichecki** [reply]
PS: If you use battery regularly, there is no point in using Conservation mode - it isn't going to magically save the battery. It is made for when you plug in the laptop and almost never unplug it.

**Reply by TheDannemand** [reply]
Thanks again, @BartoszCichecki. 

Indeed, if you and other Legion users are not having issues with the **Battery Mode** control disappearing when changing its setting, I am convinced that LLT incompatibility with my model is the cause. If I get the time one day, I may dig into the LLT log files and see if I can pinpoint it further. But I cannot recall when I last had such a day...

My laptop is plugged in all the time, except maybe once or twice a week when I use it on the patio or take it to a meeting, for which 80% charge is sufficient. Occasionally I take it on a trip or need a full charge for other reasons.

I can assure you that limiting charging to 80% for this kind of use case really DOES significantly prolong battery health, as it avoids the top and bottom 20% ranges of the battery's capacity, where most of the wear on Li-Ion batteries occurs. They wear very little in the middle range, which is why devices are often charged to 50% from the factory, knowing that they may sit on shelves for many months.

While this is my first Lenovo in decades, my previous Samsung laptop had a similar feature, which maintained the battery at over 70% health, even after 10 years of use (70%+ of design capacity). Full discharge/re-charge every six months or so is recommended to help keep battery stats accurate.

Thank again for your excellent software and for your response. Highly appreciated!
